frontend/app/work_packages/controllers/work-package-details-controller.js

```diff
@@ -172,13 +174,53 @@ module.exports = function($scope,
     return !!($scope.workPackage && $scope.workPackage.embedded.watchers !== undefined);
   };
 
-  function displayedActivities(workPackage) {
+  $scope.matchesActivityType = function(activity, type) {
+    return !!activity.props._type.match(type);
+  };
+
+  function aggregateActivities(workPackage) {
+    // Do not yet add any intermittent result to the scope,
+    // as we will get an inconsistent activity view
+    // As we may not what activities will be added at a given time,
+    // let them be aggregated asynchronously.
+    var aggregated = [],
+      totalActivities = 0;
+
+    var aggregate = function(success, activity) {
+
+      if (success === true) {
+        aggregated = aggregated.concat(activity);
+      }
+
+      if (++totalActivities === 2) {
```

Comment:
But `while (*foo++ = *bar++);` is still a quite elegant way to write a segfault ^^